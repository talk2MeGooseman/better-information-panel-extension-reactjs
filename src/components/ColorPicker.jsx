import React from 'react'
import reactCSS from 'reactcss'
import { ChromePicker } from 'react-color'
import { PropTypes } from 'prop-types';

class ColorPicker extends React.Component {
  static defaultProps = {
    color: '#000000',
  };

  static propTypes = {
    handleChange: PropTypes.func.isRequired,
  };

  state = {
    displayColorPicker: false,
    color: '#000000',
  };

  handleClick = () => {
    this.setState({ displayColorPicker: !this.state.displayColorPicker })
  };

  handleClose = () => {
    this.setState({ displayColorPicker: false })
  };

  render() {

    const styles = reactCSS({
      'default': {
        color: {
          width: '36px',
          height: '14px',
          borderRadius: '2px',
          background: `${this.props.color}`,
        },
        swatch: {
          padding: '5px',
          background: '#fff',
          borderRadius: '1px',
          boxShadow: '0 0 0 1px rgba(0,0,0,.1)',
          display: 'inline-block',
          cursor: 'pointer',
        },
        popover: {
          position: 'absolute',
          zIndex: '2',
        },
        cover: {
          position: 'fixed',
          top: '0px',
          right: '0px',
          bottom: '0px',
          left: '0px',
        },
      },
    });

    return (
      <div>
        <div className="color-swatch" style={ styles.swatch } onClick={ this.handleClick }>
          <div style={ styles.color } />
        </div>
        { this.state.displayColorPicker ? <div className="popover" style={ styles.popover }>
          <div className="closePopover" style={ styles.cover } onClick={ this.handleClose }/>
          <ChromePicker color={ this.props.color } disableAlpha={true} onChange={ this.props.handleChange } />
        </div> : null }
      </div>
    )
  }
}

export default ColorPicker
