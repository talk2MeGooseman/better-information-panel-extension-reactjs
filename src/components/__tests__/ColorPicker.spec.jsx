import React from "react";
import { shallow, mount } from "enzyme";

import ColorPicker from "../ColorPicker";

describe('ColorPicker', () => {
  let fnSpy;

  beforeEach(() => {
    fnSpy = jest.fn();
  })

  it('matches default state snapshot', () => {
    const wrapper = mount(<ColorPicker handleChange={fnSpy} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('displays prop color swatch #ffffff', () => {
    const wrapper = mount(<ColorPicker color="#ffffff" handleChange={fnSpy} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should open the ChromePicker on swatch click', () => {
    const wrapper = mount(<ColorPicker color="#ffffff" handleChange={fnSpy} />);
    expect(wrapper.find('.popover').length).toEqual(0);
    wrapper.find('.color-swatch').simulate('click');
    expect(wrapper.find('.popover').length).toEqual(1);
  });


  it('should open and close the ChromePicker onClick', () => {
    const wrapper = mount(<ColorPicker color="#ffffff" handleChange={fnSpy} />);
    expect(wrapper.find('.popover').length).toEqual(0);
    wrapper.find('.color-swatch').simulate('click');
    expect(wrapper.find('.popover').length).toEqual(1);
    wrapper.find('.closePopover').simulate('click');
    expect(wrapper.find('.popover').length).toEqual(0);
  });

});