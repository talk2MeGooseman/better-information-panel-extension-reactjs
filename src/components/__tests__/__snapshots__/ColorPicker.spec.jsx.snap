// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPicker displays prop color swatch #ffffff 1`] = `ReactWrapper {}`;

exports[`ColorPicker matches default state snapshot 1`] = `ReactWrapper {}`;
